scraper.py

```diff
@@ -0,0 +1,99 @@
+from constants import page_url, base_url, website_url
+from utils import get_soup_by_selenium_driver
+
+
+def get_product_details(soup):
+    # get product details
+    name = soup.find('div', id='scroll_car_info').get_text(separator='\n', strip=True).split('\n')[0]
+
+    price = soup.find('div', class_='price-box').get_text(separator=' ', strip=True)
+
+    location_data = (soup.find('p', class_='detail-sub-heading')
+                     .get_text(separator=',', strip=True).split(',', maxsplit=2))
+    location = ','.join(location_data[:2])
+
+    model_year = \
+        soup.find('span', class_='engine-icon year').parent.get_text(strip=True)
+
+    mileage = soup.find('span', class_='engine-icon millage').parent.get_text(strip=True)
+
+    seller_name_data = soup.find('div', class_='owner-detail-main').get_text(separator=',', strip=True).split(
+        ',')
+
+    # checking "is seller 'Dealer' or not? "
+    seller_name = seller_name_data[0]
+    if seller_name == 'Dealer:':
+        seller_name = seller_name_data[1]
+
+    featured_or_not = soup.find('div', id='myCarousel').get_text(separator=',', strip=True).split(',')[0]
+
+    is_featured = featured_or_not == 'FEATURED'
+
+    images = soup.find('div', id='myCarousel').find('ul', class_='lSPager lSGallery').findAll('img')
+
+    cover_image_url = images[0]['data-original']
+    image_1_url = images[1]['data-original'] if len(images) > 1 else None
+    image_2_url = images[2]['data-original'] if len(images) > 2 else None
+
+    return {
+        'name': name,
+        'price': price,
+        'location': location,
+        'model_year': model_year,
+        'mileage': mileage,
+        'seller_name': seller_name,
+        'is_featured': is_featured,
+        'cover_image_url': cover_image_url,
+        'image_1': image_1_url,
+        'image_2': image_2_url,
+    }
+
+
+def get_last_page():
+    soup = get_soup_by_selenium_driver(website_url)
+    last_page_url_data = soup.find('li', class_='last next')
+
+    if last_page_url_data:
+        last_page_url = last_page_url_data.find('a')['href']
+        last_page_url = base_url + last_page_url
+        return last_page_url
+    else:
+        return None
+
+
+def get_page_urls():
+    page_num = 1
+    page_urls_list = []
+    last_page_url = get_last_page()
+
+    if last_page_url:
+        while last_page_url not in page_urls_list:
+            page_urls_list.append(f'{base_url}{page_url}?page={page_num}')
+            page_num += 1
+    else:
```

Comment:
use this logic:
1. find the last page index number by using regex
2. iterate from 1 to last page's index which you found from step1
3. append everything in a list